app/assets/javascripts/discourse/app/controllers/user-private-messages.js

```diff
@@ -3,26 +3,118 @@ import { action } from "@ember/object";
 import { alias, and, equal } from "@ember/object/computed";
 import discourseComputed from "discourse-common/utils/decorators";
 import { VIEW_NAME_WARNINGS } from "discourse/routes/user-private-messages-warnings";
+import I18n from "I18n";
+
+export const personalInbox = "__personal_inbox__";
+const allInbox = "__all_inbox__";
 
 export default Controller.extend({
+  queryParams: ["tag"],
   user: controller(),
 
   pmView: false,
   viewingSelf: alias("user.viewingSelf"),
   isGroup: equal("pmView", "groups"),
+  group: null,
+  groupFilter: alias("group.name"),
   currentPath: alias("router._router.currentPath"),
   pmTaggingEnabled: alias("site.can_tag_pms"),
-  tagId: null,
+  tag: null,
 
   showNewPM: and("user.viewingSelf", "currentUser.can_send_private_messages"),
 
+  @discourseComputed("pmView")
+  isPersonalInbox(pmView) {
+    return pmView && pmView.startsWith("personal");
+  },
+
+  @discourseComputed("isPersonalInbox", "group.name")
+  isAllInbox(isPersonalInbox, groupName) {
+    return !this.isPersonalInbox && !groupName;
+  },
+
+  @discourseComputed("isPersonalInbox", "group.name")
+  selectedInbox(isPersonalInbox, groupName) {
+    if (groupName) {
+      return groupName;
+    }
+
+    return isPersonalInbox ? personalInbox : allInbox;
+  },
+
   @discourseComputed("viewingSelf", "pmView", "currentUser.admin")
   showWarningsWarning(viewingSelf, pmView, isAdmin) {
     return pmView === VIEW_NAME_WARNINGS && !viewingSelf && !isAdmin;
   },
 
+  @discourseComputed("tags")
+  tagsContent(tags) {
+    return tags.map((tag) => {
+      return { id: tag.id, name: tag.text };
+    });
+  },
+
+  @discourseComputed("siteSettings.tagging_enabled", "tagsContent")
+  displayTagDropdown(taggingEnabled, tagsContent) {
+    return taggingEnabled && tagsContent.length !== 0;
+  },
+
+  @discourseComputed("inboxes")
+  displayInboxDropdown(inboxes) {
```

Comment:
Ditto here, this could be `{{#if inboxes}}` (or perhaps `{{#if inboxes.length}}`)